views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -73,48 +78,136 @@ define([
-        $responseForm.html(addAnswerOptionBtn());
-        this.initAddAnswerButton();
-        this.renderForm(self.correctResponses);
+        this.correctResponses = this.getExistingCorrectAnswerOptions();
+        $responseForm.html(responseFormTpl({
+            identifier: interaction.attr('responseIdentifier'),
+            serial: response.serial,
+            min: interaction.prop('min'),
+            max: interaction.prop('max'),
+            mappingDisabled,
+            defaultValue: response.getMappingAttribute('defaultValue')
+        }));
+
+        minMaxComponentFactory($responseForm.find('.response-mapping-attributes > .min-max-panel'), {
+            min: {
+                fieldName: 'lowerBound',
+                value: _.parseInt(response.getMappingAttribute('lowerBound')) || 0,
+                helpMessage: __('Minimal  score for this interaction.')
+            },
+            max: {
+                fieldName: 'upperBound',
+                value: _.parseInt(response.getMappingAttribute('upperBound')) || 0,
+                helpMessage: __('Maximal score for this interaction.')
+            },
+            upperThreshold: Number.MAX_SAFE_INTEGER,
+            syncValues: true
+        });     
+        this.initResponseForm();
+        this.createScoreResponse();
+        this.initEditingOptions();
+
+        this.initAlternativeInput();
     }
 
-    MathEntryInteractionStateResponse.prototype.initAddAnswerButton = function initAddAnswerButton() {
-        var self = this,
-            interaction = self.widget.element,
-            $responseForm = self.widget.$responseForm,
-            $addAnswerBtn = $responseForm.find($('.add-answer-option'));
-
-        $addAnswerBtn.on('click', function () {
-            var newCorrectAnswer;
-
-            if (self.inGapMode() === true) {
-                self.emptyGapFields();
-                var gapExpression = interaction.prop('gapExpression');
-                var gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
-                if (gapCount > 0) {
-                    newCorrectAnswer = [];
-                    for (var i = 0; i < gapCount; i++) {
-                        newCorrectAnswer.push(' ');
-                    }
+    MathEntryInteractionStateResponse.prototype.initResponseForm = function initResponseForm() {
+        if (this.correctResponses.length > 0) {
+            return false;
+        }
+        var interaction = this.widget.element;
 
-                    newCorrectAnswer = newCorrectAnswer.join(',');
-                } else {
-                    newCorrectAnswer = '';
+        var newCorrectAnswer;
+
+        if (this.inGapMode() === true) {
+            this.emptyGapFields();
+            var gapExpression = interaction.prop('gapExpression');
+            var gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
+            if (gapCount > 0) {
+                newCorrectAnswer = [];
+                for (var i = 0; i < gapCount; i++) {
```

Comment:
**issue:** please don't use `var` this way. Remember the [hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting). Variables declared with `var` must be declared on top. But as ES6+ is now activated for PCI, please rather use `const` and `let`